### DISCUSSION TITLE
Scope for the new Flows system

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#14852] [Archive]
We have an small hook custom extension like this:

```javascript
const bson = require('bson');

module.exports = ({ filter }) => {
  filter('battles.items.create', async (input) => {
    return {
      ...input,
      id: new bson.ObjectID().toString(),
    };
  });
};
```

And another like this:

```javascript
const slugify = require('@sindresorhus/slugify');

module.exports = ({ filter }) => {
  filter('competitions.items.create', async (input) => {
    return {
      ...input,
      slug: slugify(input.name),
    };
  });

  filter('competitions.items.update', async (input) => {
    if (input.name) {
      return {
        ...input,
        slug: slugify(input.name),
      };
    }
  });
};
```

This can't be done with a Flow, right? And do you plan to be able to do it?
I imagine not, but to confirm.

**Comment by rijkvanzanten** [comment]
Beyond creating a slugify / bson custom operation extension, or using a webhook to an external service to do this, this can't be done in flows currently natively. That being said, we're currently working on https://github.com/directus/directus/discussions/13944, which adds a way to run some arbitrary JS in an operation. That most likely won't be able to use custom third party node modules though, so I don't think it would cover your use case exactly 🤔
